scripts/Dockerfile

```diff
@@ -16,54 +15,34 @@ WORKDIR /home/pybamm/
 RUN git clone https://github.com/pybamm-team/PyBaMM.git
 
 WORKDIR /home/pybamm/PyBaMM
+USER root
+
+# .local not present by default
+RUN mkdir /home/pybamm/.local
 
 ENV CMAKE_C_COMPILER=/usr/bin/gcc
 ENV CMAKE_CXX_COMPILER=/usr/bin/g++
 ENV CMAKE_MAKE_PROGRAM=/usr/bin/make
 ENV SUNDIALS_INST=/home/pybamm/.local
 ENV LD_LIBRARY_PATH=/home/pybamm/.local/lib
 
-RUN conda create -n pybamm python=3.11
-RUN conda init --all
-SHELL ["conda", "run", "-n", "pybamm", "/bin/bash", "-c"]
-RUN conda install -y pip
+# Migrating to venv and create a virtual environment for python 3.11
+ENV VIRTUAL_ENV=/home/pybamm/venv
+RUN python3.11 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
-ARG IDAKLU
-ARG ODES
-ARG JAX
-ARG ALL
-
-RUN pip install --upgrade --user pip setuptools wheel wget
+RUN pip install --upgrade setuptools wheel wget
 RUN pip install cmake
+RUN source /home/pybamm/venv/bin/activate
 
-RUN if [ "$IDAKLU" = "true" ]; then \
-    python scripts/install_KLU_Sundials.py && \
-    rm -rf pybind11 && \
-    git clone https://github.com/pybind/pybind11.git && \
-    pip install --user -e ".[all,dev,docs]"; \
-    fi
-
-RUN if [ "$ODES" = "true" ]; then \
-    python scripts/install_KLU_Sundials.py && \
-    pip install --user -e ".[all,dev,docs,odes]"; \
-    fi
+RUN python scripts/install_KLU_Sundials.py --install-dir /home/pybamm/.local && \
+python scripts/install_KLU_Sundials.py && \
+mv /home/pybamm/.local/lib64 /home/pybamm/.local/lib && \
+rm -rf pybind11 && \
+git clone https://github.com/pybind/pybind11.git && \
+pip install -e ".[all,dev,docs,jax,odes]";
 
-RUN if [ "$JAX" = "true" ]; then \
-    pip install --user -e ".[all,dev,docs,jax]"; \
-    fi
-
-RUN if [ "$ALL" = "true" ]; then \
-    python scripts/install_KLU_Sundials.py && \
-    rm -rf pybind11 && \
-    git clone https://github.com/pybind/pybind11.git && \
-    pip install --user -e ".[all,dev,docs,jax,odes]"; \
-    fi
-
-RUN if [ -z "$IDAKLU" ] \
-    && [ -z "$ODES" ] \
-    && [ -z "$JAX" ] \
```

Comment:
These options should not have been removed. Was there a reason for removing them?

Reply:
We decided to switch to a single Docker image containing all solvers in the last developer meeting on 19/02/2024 – please see #3666 for context

Reply:
These options have been removed to perform a full installation with all the solvers inside the container